crates/ty_python_semantic/resources/mdtest/snapshots/no_matching_overload…_-_No_matching_overload…_-_Call_to_function_wit…_(dd80c593d9136f35).snap

```diff
@@ -0,0 +1,59 @@
+---
+source: crates/ty_test/src/lib.rs
+expression: snapshot
+---
+---
+mdtest name: no_matching_overload.md - No matching overload diagnostics - Call to function with many unmatched overloads
+mdtest path: crates/ty_python_semantic/resources/mdtest/diagnostics/no_matching_overload.md
+---
+
+# Python source files
+
+## mdtest_snippet.py
+
+```
+1 | class Foo: ...
+2 | 
+3 | pow(Foo(), Foo())  # error: [no-matching-overload]
+```
+
+# Diagnostics
+
+```
+error[no-matching-overload]: No overload of function `pow` matches arguments
+ --> src/mdtest_snippet.py:3:1
+  |
+1 | class Foo: ...
+2 |
+3 | pow(Foo(), Foo())  # error: [no-matching-overload]
+  | ^^^^^^^^^^^^^^^^^
+  |
+info: First overload defined here
+    --> stdlib/builtins.pyi:1700:8
+     |
+1698 | # but adding a `NoReturn` overload isn't a good solution for expressing that (see #8566).
+1699 | @overload
+1700 | def pow(base: int, exp: int, mod: int) -> int: ...
+     |        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

Comment:
nit: this would look more natural to me if the underline covered the function name too

```suggestion
1700 | def pow(base: int, exp: int, mod: int) -> int: ...
     |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

Reply:
I don't see a way to get a single span covering both of those. At least, it's not `StmtFunctionDef`: https://github.com/astral-sh/ruff/blob/97d7b4693684753139b15f86bc796df9ee25e76e/crates/ruff_python_ast/src/generated.rs#L6606-L6616

I can annotate _both_ the name and the parameter list, which will probably look fine in most cases, but could be a little weird if and when there is spacing between the name and the start of the parameter list.

Reply:
can't you just do something similar to what we do to get the "header span" for classes?

https://github.com/astral-sh/ruff/blob/97d7b4693684753139b15f86bc796df9ee25e76e/crates/ty_python_semantic/src/types/class.rs#L1835-L1862

Reply:
You need to construct a new Text range starting at `name.start` and ending at `parameters.end`

Reply:
I guess that works. I'm always cautious about stitching together spans, but I think it works here.